tripy/tripy/frontend/module/batchnorm.py

```diff
@@ -0,0 +1,116 @@
+#
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from dataclasses import dataclass
+
+from tripy import export, utils
+from tripy.common import datatype
+from tripy.frontend.module.module import Module
+from tripy.frontend.module.parameter import DefaultParameter, Parameter
+
+@export.public_api(document_under="operations/modules")
+@dataclass
+@utils.constant_fields(["num_features"])
+class BatchNorm(Module):
+    r"""
+    Applies batch normalization over an N-dimensional input tensor using precomputed statistics.
+
+    This implementation supports 1D, 2D, and 3D inputs (e.g., time-series, images, and volumetric data).
+    Batch Normalization normalizes across the specified feature dimension (typically the second dimension in the input).
+    The mean (:math:`\mu`) and variance (:math:`\sigma^2`) used are the precomputed running statistics.
+
+    :math:`y = \frac{x - \mu}{\sqrt{\sigma^2 + \epsilon}} * \gamma + \beta`
+
+    where:
+        - :math:`\mu` is the precomputed running mean.
+        - :math:`\sigma^2` is the precomputed running variance.
+        - :math:`\gamma` and :math:`\beta` are learnable parameter vectors (wieight and bias).
+
+    This module is designed for evaluation purposes only, and it does not compute batch statistics.
+    """
+
+    num_features: int
+    r"""The number of feature channels in the input tensor (the size of the second dimension)."""
+
+    weight: Parameter
+    r"""The learnable scale parameter (γ) of shape `(num_features,)`."""
+
+    bias: Parameter
+    r"""The learnable shift parameter (β) of shape `(num_features,)`."""
+
+    running_mean: Parameter
+    r"""The precomputed running mean for the feature channels of shape `(num_features,)`."""
+
+    running_var: Parameter
+    r"""The precomputed running variance for the feature channels of shape `(num_features,)`."""
```

Comment:
Everything here is precomputed, so we can omit that:
```suggestion
    r"""The running mean for the feature channels of shape :math:`[\text{num_features}]`."""

    running_var: Parameter
    r"""The running variance for the feature channels of shape :math:`[\text{num_features}]`."""
```